lib/ljsyscall/syscall/linux/syscalls.lua

```diff
@@ -457,10 +457,42 @@ function S.sched_setaffinity(pid, mask, len) -- note len last as rarely used
   return retbool(C.sched_setaffinity(pid or 0, len or s.cpu_set, mktype(t.cpu_set, mask)))
 end
 
+local function get_maxnumnodes()
+   local function filesize(fd)
+      local size = 0
+      local bufsz = 1024
+      local buf = ffi.new("uint8_t[?]", bufsz)
+      while true do
+         local len, err = S.read(fd, buf, bufsz)
+         size = size + len
+         if len ~= bufsz then break end
+      end
+      fd:seek(0, "set")
+      return size
+   end
+   local function read(filename)
+      local fd, errno = S.open(filename, 0)
+      if not fd then error(errno) end
+      local content, errno = S.read(fd, nil, filesize(fd))
+      if not content then error(errno) end
+      fd:close()
+      return content
+   end
+   local content = read("/proc/self/status")
+   for line in content:gmatch("[^\n]+") do
+     if line:match("^Mems_allowed:") then
+       line = line:gsub("^Mems_allowed:%s+", "")
```

Comment:
This needs a comment about what's going on here; why the 9 (I know, because NNNNNNNN,), why the 32.